mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala

```diff
@@ -165,13 +165,15 @@ class BinaryClassificationMetrics @Since("3.0.0") (
     confusions: RDD[(Double, BinaryConfusionMatrix)]) = {
     // Create a bin for each distinct score value, count weighted positives and
     // negatives within each bin, and then sort by score values in descending order.
-    val counts = scoreLabelsWeight.combineByKey(
+    val binnedWeights = scoreLabelsWeight.combineByKey(
       createCombiner = (labelAndWeight: (Double, Double)) =>
         new BinaryLabelCounter(0.0, 0.0) += (labelAndWeight._1, labelAndWeight._2),
       mergeValue = (c: BinaryLabelCounter, labelAndWeight: (Double, Double)) =>
         c += (labelAndWeight._1, labelAndWeight._2),
       mergeCombiners = (c1: BinaryLabelCounter, c2: BinaryLabelCounter) => c1 += c2
-    ).sortByKey(ascending = false)
+    )
+    binnedWeights.persist()
```

Comment:
Could you explain why do you persist `binnedWeights` but not `counts`? As I can see `binnedWeights` is used only once ... or I missed something?

Reply:
Rdd `scoreAndLabels.combineByKey` is found to be used more than one time, as mentioned in [SPARK-29816](https://issues.apache.org/jira/browse/SPARK-29816).
I think, `counts` rdd is constructed over `scoreAndLabels.combineByKey` after action `sortByKey`. `counts` is indirectly dependent on `scoreAndLabels.combineByKey` i.e, `binnedWeights`?